lib/diaspora_federation/properties_dsl.rb

```diff
@@ -71,11 +71,18 @@ def class_prop_names
     def define_property(name, type, opts={})
       raise InvalidName unless name_valid?(name)
 
-      xml_name = name
       if opts.has_key? :xml_name
         raise ArgumentError, "xml_name is not supported for nested entities" unless type == String
         xml_name = opts[:xml_name]
         raise InvalidName, "invalid xml_name" unless name_valid?(xml_name)
+      elsif type == String
+        xml_name = name
+      elsif type.instance_of?(Array)
+        xml_name = type.first.entity_name.to_sym
+      elsif type.ancestors.include?(Entity)
+        xml_name = type.entity_name.to_sym
+      else
+        raise ArgumentError, "unknown type #{type} supplied"
```

Comment:
can you extract the xml_name stuff to its own method (because rubocop says: Assignment Branch Condition size for define_property is too high.) and maybe merge the existing xml_name logic with the `if type == String` part.